src/box/lua/connpool.lua

```diff
@@ -25,13 +25,58 @@ local function is_connection_valid(conn, opts)
     return true
 end
 
+function pool_methods._unused_connection_watchdog_step(self)
+    local now = fiber.clock()
+    local until_next_deadline = math.huge
+
+    for name, conn in pairs(self._connections) do
+        local until_deadline = conn._deadline - now
+
+        if until_deadline <= 0 then
+            if is_connection_valid(conn, {}) then
+                conn:close()
+            end
+
+            self._connections[name] = nil
+        elseif until_deadline < until_next_deadline then
+            until_next_deadline = until_deadline
+        end
+    end
+
+    return until_next_deadline
+end
+
+function pool_methods._unused_connection_watchdog_loop(self)
+    while true do
+        local until_next_deadline = self:_unused_connection_watchdog_step()
+
+        if until_next_deadline == math.huge then
+            break
+        end
+
+        self._unused_connection_watchdog_cond:wait(until_next_deadline)
+    end
+end
+
+function pool_methods._unused_connection_watchdog_wake(self)
+    local f = self._unused_connection_watchdog_fiber
+    if f ~= nil and f:status() ~= 'dead' then
+        self._unused_connection_watchdog_cond:broadcast()
```

Comment:
It seems that there is only one fiber that waits the condition. Is it correct? If so, why `:broadcast()` is used, not just `:signal()`?

If there are some nuances that I don't see, it worth to leave a remark.